File: src/idleTimer.ts
export class IdleTimer {
  private timeout: NodeJS.Timeout;
  private timeoutCallback: () => void;
  private timeoutTime: number;

  constructor(fn: () => void, timeoutValue: number) {
    this.timeout = setTimeout(fn, timeoutValue);
    this.timeoutCallback = fn;
    this.timeoutTime = timeoutValue;
  }

  public resetTimeout = () => {
    clearTimeout(this.timeout);
    this.timeout = setTimeout(this.timeoutCallback, this.timeoutTime);
  };
}

File: CSharp/ChatApp/IdleTimer.cs
using System;

namespace ChatApp
{
    public class IdleTimer
    {
        private readonly Action timeoutCallback;
        private readonly int timeoutTime;
        private System.Threading.Timer timeout;

        public IdleTimer(Action fn, int timeoutValue)
        {
            timeoutCallback = fn;
            timeoutTime = timeoutValue;
            timeout = new System.Threading.Timer(_ => fn(), null, timeoutValue, System.Threading.Timeout.Infinite);
        }

        public void ResetTimeout()
        {
            timeout.Change(timeoutTime, System.Threading.Timeout.Infinite);
        }
    }
}